power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -350,6 +354,79 @@ template <bool sym> class YBus {
         admittance_ = std::make_shared<ComplexTensorVector<sym> const>(std::move(admittance));
     }
 
+    template <UpdateIsProgressive T>
+    void update_admittance_increment(std::shared_ptr<T const> const& math_model_param_incrmt,
+                                     bool is_decrement = false) {
+        const double mode = is_decrement ? -1.0 : 1.0;
+        if (!is_decrement) {
+            // overwrite the old cached parameters in increment mode (update)
+            math_model_param_incrmt_ = math_model_param_incrmt;
+        } else if (!decremented_) {
+            math_model_param_incrmt_ = {}; // reset recorded increment that is already decremented
+            return;
+        }
+
+        // construct admittance data
+        ComplexTensorVector<sym> admittance(nnz());
+        assert(Idx(admittance_->size()) == nnz());
+        std::ranges::copy(admittance_->begin(), admittance_->end(), admittance.begin());
+        auto const& b_param = math_model_param_incrmt_->branch_param;
+        auto const& b_param_2c = math_model_param_incrmt_->branch_param_to_change;
+        auto const& s_param = math_model_param_incrmt_->shunt_param;
+        auto const& s_param_2c = math_model_param_incrmt_->shunt_param_to_change;
+        auto const& y_bus_element = y_bus_struct_->y_bus_element;
+        auto const& y_bus_entry_indptr = y_bus_struct_->y_bus_entry_indptr;
+        auto const& y_bus_pos_in_entries = y_bus_struct_->y_bus_pos_in_entries;
+
+        // construct affected entries
+        std::vector<Idx> affected_entries = {};
+
+        auto check_and_add = [&affected_entries, &y_bus_pos_in_entries](MatrixPos pos) {
+            auto it = std::ranges::find(y_bus_pos_in_entries.begin(), y_bus_pos_in_entries.end(), pos);
+            if (it != y_bus_pos_in_entries.end()) {
+                affected_entries.push_back(std::distance(y_bus_pos_in_entries.begin(), it));
+            }
+        };
+
+        auto params_to_change = {b_param_2c | std::views::transform([](int n) { return std::make_pair(n, 1); }),
+                                 s_param_2c | std::views::transform([](int n) { return std::make_pair(n, 2); })};
+
+        for (auto [param_to_change, ptype] : params_to_change) {
+            Idx id_x{0};
+            Idx id_y{0};
+            if (ptype == 1) {
+                id_x = math_topology_->branch_bus_idx[param_to_change][0];
+                id_y = math_topology_->branch_bus_idx[param_to_change][1];
+            } else {
+                id_x = math_topology_->shunts_per_bus.get_group(param_to_change);
+                id_y = math_topology_->shunts_per_bus.get_group(param_to_change);
+            }
+            check_and_add(std::pair(id_x, id_y));
```

Comment:
why do you need a pair?

if you really need it, you can do either

```suggestion
            check_and_add({id_x, id_y});
```

or

```suggestion
            check_and_add(std::pair{id_x, id_y});
```

or

```suggestion
            check_and_add(std::make_pair(id_x, id_y));
```

using constructors with braces `(...)` is actually considered bad practice (see e.g. https://stackoverflow.com/questions/18222926/what-are-the-advantages-of-list-initialization-using-curly-braces , it's a bit old but it does explain the main issues)

Reply:
Changing it from `std::pair` to `MatrixPos`

Reply:
or just have the function itself ingest a two values `check_and_add(id_x, id_y)`. but are you sure that `check_and_add({id_x, id_y})` doesn't work?

Reply:
function refactored